### ISSUE TITLE
Bridgy is somehow getting incorrect URLs for webmentions of my site

### ISSUE THREAD

**Issue by mbifulco** [snarfed/bridgy#946]
Crossposting this from an issue I started with webmention.io [here](https://github.com/aaronpk/webmention.io/issues/154)

It looks like Bridgy is trying to send notifications from twitter to bad URLs from my site (mikebifulco.com) - it keeps sending off things that look like this:
`https://mikebifulco.com/why-fathom-analytics,%20/why-fathom-analytics/` - with an extra `,%20/why-fathom-analytics` at the end. Have you seen that before? I'm at a loss for what might cause this

Screenshot:
![image](https://user-images.githubusercontent.com/1844496/82737349-8843ea80-9cfe-11ea-9439-399523d918b3.png)

This is the twitter thread for that particular notification: https://twitter.com/irreverentmike/status/1264207418759753731

**Comment by aaronpk** [comment]
Posting my comment from the other thread: https://github.com/aaronpk/webmention.io/issues/154#issuecomment-633110808

**Comment by aaronpk** [comment]
I suspect I know what's going on here...

I'm guessing the python `requests` client is finding the two `Location` headers being returned in the response, and combining the values as a comma-separated string the way it would for other repeated HTTP headers.

**Comment by snarfed** [comment]
huh. actually i'm seeing that https://www.mikebifulco.com/why-fathom-analytics results in _two_ redirects:
1. 301 to `https://mikebifulco.com/why-fathom-analytics`, which then returns
1. 301 to `/why-fathom-analytics/`,
1. which 200s.

same in firefox, curl, and python requests. i don't see multiple `Location` headers in either of those two redirects.

**Comment by snarfed** [comment]
(you're right that requests does concatenate multiple header values with comma, but that's not happening here, at least in testing from my laptop.)

**Comment by aaronpk** [comment]
Sorry yeah that's what I mean, I described it better on the webmention.io issue linked. Because the page `https://mikebifulco.com/why-fathom-analytics` returns a Location header *as well* as the contents of the page, I suspect that's what's confusing the HTTP client and causing it to treat that last `/why-fathom-analytics/` as part of the same response, then combining the two `Location` headers into one.

**Comment by snarfed** [comment]
...having said that, you're right that _something_ somewhere along the line is comma-concatenating the two redirects' header values. python requests does seem like the most likely, but i tested manually just now, and it didn't. so the plot thickens.

**Comment by snarfed** [comment]
another useful data point: bridgy successfully sent a webmention to another post on this site yesterday, https://www.mikebifulco.com/deconfusing-javascript-destructuring-syntax . the tweet, https://twitter.com/irreverentmike/status/1263837389321113605 , points to the same www type URL (above), which does the same double redirect. so i suspect something unusual is going on with either this specific post or the specific webmention.

the successful [bridgy log](https://brid.gy/log?start_time=1590169217&key=agdicmlkLWd5cjYLEghSZXNwb25zZSIodGFnOnR3aXR0ZXIuY29tLDIwMTM6MTI2Mzg3NjY2MjQ5Nzg2MTY0Mww) looks pretty much the same:
```
Params: [('response_key', Params: [('response_key', 'agdicmlkLWd5cjYLEghSZXNwb25zZSIodGFnOnR3aXR0ZXIuY29tLDIwMTM6MTI2Mzg3NjY2MjQ5Nzg2MTY0Mww')])]
Source: irreverentmike (Twitter) irreverentmike, https://brid.gy/twitter/irreverentmike
Created by this poll: https://brid.gy/log?start_time=1590169154&key=agdicmlkLWd5chsLEgdUd2l0dGVyIg5pcnJldmVyZW50bWlrZQw
Starting Response comment tag:twitter.com,2013:1263876662497861643 https://twitter.com/KentorThames/status/1263876662497861643
Webmention from https://brid-gy.appspot.com/comment/twitter/irreverentmike/1263837389321113605/1263876662497861643 to https://www.mikebifulco.com/deconfusing-javascript-destructuring-syntax/
Sending...
Sent! {'request': 'POST https://webmention.io/mikebifulco.com/webmention (with source=https://brid-gy.appspot.com/comment/twitter/irreverentmike/1263837389321113605/1263876662497861643, target=https://www.mikebifulco.com/deconfusing-javascript-destructuring-syntax/)', 'http_status': 201, 'body': '{"status":"queued","summary":"Webmention was queued for processing","location":"https://webmention.io/mikebifulco.com/webmention/Fky0QJ5cyDSgPXK5Gqfr","source":"https://brid-gy.appspot.com/comment/twitter/irreverentmike/1263837389321113605/1263876662497861643","target":"https://www.mikebifulco.com/deconfusing-javascript-destructuring-syntax/"}'}
Setting last_webmention_sent
```

**Comment by mbifulco** [comment]
Thanks for all the help you two - I _thought_ I had a relatively simple site setup, but something is clearly wrong on my end. Investigating...

**Comment by mbifulco** [comment]
Okay, a couple updates - the tweet I linked above was _before_ I switched the site from forcing `www.` to _not_ forcing `www.` - so I think the t.co link was an old one.  Responses to newer actions on links from twitter seem to be better:

This [tweet](https://twitter.com/irreverentmike/status/1264294877766070272) currently has 1 like, and it is showing up in bridgy as I'd expect:

![image](https://user-images.githubusercontent.com/1844496/82740277-5b4f0200-9d15-11ea-8caa-e868290a84c2.png)

And subsequently on webmention.io
![image](https://user-images.githubusercontent.com/1844496/82740291-70c42c00-9d15-11ea-8937-30ae06434b53.png)

So I've got functionality back on my site, which is great!  I am still seeing a 301 redirect on page loads, so I've got to do some spelunking to figure out what that's about.  It may be a Gatsby setup issue, or something with netlify, _or_ something with my DNS setup on cloudflare.  I'll report back with when I figure out more.

Thank you both for all your help so far -- you've gone above and beyond 🙏
